### DISCUSSION TITLE
Is it possible ? Background Service only (without being recognized as Mini App)

### DISCUSSION THREAD

**Discussion by tip-n** [zepp-health/zeppos-forum#187] [Q&A] [answered]
Is it possible with v1 or v2 to create a background service which run at interval time without it being an app or need a trigger like button ?

I've taken a look at https://docs.zepp.com/docs/guides/framework/side-service/register/
but I didn't found anything about interval or time related in side service

example : 
- I want to make an auto detect background service which will run a function every minutes without button to trigger the function
- I want to make an background service which will fetch api and synchronize with storage in smartwatch every minutes

I've tried using : 
- timer (fail, timer = undefined because it's placed in AppSideService and not in Watchface)
```
      timer.createTimer(500, 1000, () => {
        conole.log("pass")
      })
```
- time.onPerMinute (fail, not running)
```
    const time = new Time()
    const onMinutePasssed = () => {
      console.log("minute passed !")
    }
    time.onPerMinute(onMinutePasssed)
    
```

Conclusion : I want to make a SideAppService only which run in the background and not an application, is it possible ?

**Comment by MyoungXUE** [comment, marked as answer]
Hello

I'm sorry, we don't have the ability to run in the background on Zepp OS V1 and V2, but your idea is very good, we have planned the concept of Background Service in Zepp OS V3, you can learn more about it here 

https: https://docs-testing.zepp.com/docs/next/guides/framework/device/background-service/

**Reply by tip-n** [reply]
Thanks for the reply 👍
